mountpoint-s3/src/fstab.rs

```diff
@@ -64,36 +68,55 @@ impl FsTabCliArgs {
         }
     }
 
-    fn validate_options(options: &[String]) -> Result<(), clap::Error> {
+    fn validate_options(options: &[String], prefix_allowed: bool) -> Result<(), clap::Error> {
         if options.iter().any(|x| x.starts_with("-")) {
-            return Err(Command::error(
-                &mut Self::command(),
+            return Self::command_error(
                 ErrorKind::InvalidValue,
                 "Cannot prefix arguments with `--` when using fstab style arguments.",
-            ));
+            );
         }
         if options.iter().any(|x| x == "ro") && options.iter().any(|x| x == "rw") {
-            return Err(Command::error(
-                &mut Self::command(),
+            return Self::command_error(
                 ErrorKind::ArgumentConflict,
                 "Cannot use 'ro' flag combined with 'rw' flag.",
-            ));
+            );
         }
-        if let Some(disallowed_option) = options
-            .iter()
-            .find(|x| ["foreground", "read-only"].contains(&x.as_str()))
-        {
-            return Err(Command::error(
-                &mut Self::command(),
+        if options.iter().any(|x| x == "foreground") {
+            return Self::command_error(
                 ErrorKind::InvalidValue,
-                format!(
-                    "Cannot use '{}' flag when using fstab style arguments.",
-                    disallowed_option
-                ),
-            ));
+                "Cannot use 'foreground' flag when using fstab style arguments.",
+            );
+        }
+        if options.iter().any(|x| x == "read-only") {
+            return Self::command_error(
+                ErrorKind::InvalidValue,
+                "Cannot use 'read-only' flag when using fstab style arguments. Use 'ro' instead",
+            );
+        }
+        if !prefix_allowed && options.iter().any(|x| x == "prefix") {
+            return Self::command_error(
+                ErrorKind::ArgumentConflict,
+                "Cannot use 'prefix' flag when using s3:// URIs for device in fstab",
```

Comment:
"... s3:// URIs with prefix ..." maybe? As this would be allowed if you don't have any prefix in s3 URI if I understand correctly